ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -29,24 +31,30 @@ def attach_context(request):
         service = app[CONFIG_KEY]['service']
         distributed_tracing = app[CONFIG_KEY]['distributed_tracing_enabled']
 
+        context = tracer.get_call_context()
+
+        # Create a new context based on the propagated information
+        # Do not fill context with distributed sampling if the tracer is disabled
+        # because the would call the callee to generate references to data which
+        # has never been sent to agent.
+        if tracer.enabled and distributed_tracing:
```

Comment:
I'd suggest to add a TODO here.

We could clearly have a helper somewhere which does http_headers -> new_context. As this will be reusable for all HTTP-related libs.